Landing-Page Site and Versioned Docs Site

samredai opened the pull request:
This addresses 3 main issues raised in #1.

- Broken Links
  - All links have been fixed and verified with https://github.com/filiph/linkcheck
- Split the landing page and the docs site with only the docs site versioned
  - The deployment workflow has two jobs, `deploy-landing-page` and `deploy-docs`
  - Commits to `main` deploys **only** the single landing page
  - Commits to any non-`main` branch deploys **only** a docs site version (explained in the README)
- Too many items in the navbar
  - The navbar has been shortened by hidding some pages and making them available in consolidated single pages with a list of links. See the `About` and `ASF` sections.
  - This is completely flexible and so any pages can be exposed in any of the following 3 places
    1. Landing Page
    2. Left Navbar
    3. Within a Page View
    - (Let me know if any item in one of these places feels like it should be in another place)

I have all of this running in my fork at:
~https://samredai.github.io/iceberg-docs~
https://iceberg.redai.dev/

To check for broken links I ran these commands:
```
docker run --rm tennox/linkcheck https://samredai.github.io/iceberg-docs > landing-page-linkcheck.txt
```
```
docker run --rm tennox/linkcheck https://samredai.github.io/iceberg-docs/docs/0.12.1 > 0.12.1-docsite-linkcheck.txt
```
```
docker run --rm tennox/linkcheck https://samredai.github.io/iceberg-docs/docs/0.12.0 > 0.12.0-docsite-linkcheck.txt
```

There are a number of broken links that show up but they're all links to the javadocs (cat command into `grep -v 'javadoc'` filters out all of the broken links detected). Once we add the javadocs I can address those.

rdblue commented:
Here are a few notes from clicking around:
* Is the final path going to be `iceberg.apache.org/iceberg-docs`? Why not make the landing page at the root level?
* In the preview, the "Learn More" link is broken. It points to https://samredai.github.io/docs instead of https://samredai.github.io/iceberg-docs/docs/, which is where the docs link takes you.
* Since we also have a new front page, what about having the Docs link send you to a Getting Started page?
* Clicking on "Tables" in the docs left nav is an empty page. Maybe we can make that a set of links to learn more about general table topics?
* Same with Integrations, API, and Format. The text is a link, but there's no page there.
* Under releases, should we include "latest"?

jackye1995 commented:
Also tried the website, thanks for continuing the work!

It seems like some links are still broken as Ryan points out. I have some additional nitpicking:

1. Github and Slack buttons should open in a new page instead of loading in the current page.
2. the time travel script is setting a table property to perform time-travel, can we use the latest time travel syntax `AS OF`?
3. for the issue that there is no link for those top level tabs like "Integration", I think we can just disable the link? Because the 2 that has link (Spark and Flink) are all linking to the getting started page, I think we can make them just a new sub-tab called `Getting started` and disable all top level tab links.
4. Once I am in the doc page, I cannot go back to the landing page anymore, can we at least have a way to go back?
5. also agree that feature list is not full, we can think about the list of features together

samredai commented:
Thanks for all the feedback! Here's an updated demo site to show all the fixes: https://iceberg.redai.dev/

@rdblue 
Is the final path going to be iceberg.apache.org/iceberg-docs? Why not make the landing page at the root level?

 - No there won't be any "/iceberg-docs". This is mainly because I'm using the stock personal gh-pages without a custom domain so it appends the repo name. I had a domain
lying around that just plugged into this demo (once it's merged we'll be able to do "dev" deployments of the site to something like "/unstable"). So now the demo urls will match how the site will look, besides the domain of course. (see https://iceberg.redai.dev/)

In the preview, the "Learn More" link is broken. It points to https://samredai.github.io/docs instead of https://samredai.github.io/iceberg-docs/docs/, which is where the docs link takes you.
 - This is slightly related to the previous item and is also fixed.

Since we also have a new front page, what about having the Docs link send you to a Getting Started page?
 - I like it! I changed Learn More to link to the spark getting-started page (we can update this as we add more getting started material)

Clicking on "Tables" in the docs left nav is an empty page. Maybe we can make that a set of links to learn more about general table topics?
  - Removed the empty link, and that's a great idea. When we add the new set of getting started material I'll re-link this section titles.

Same with Integrations, API, and Format. The text is a link, but there's no page there.
 - Fixed by removing the link

Under releases, should we include "latest"?
 - I was overthinking this a bit and removed latest because it's a clone of the 0.12.1 site. Thinking about it now it actually makes sense to keep them as two different sites so I updated this. Also included a link to the "Release Notes" page as an item instead of linking the section title which wasn't obvious.

The About page is odd and just links to other pages
 - Fixed when changing the left nav to collapsible sections at the bottom

Looks like the ASF and About nav pages are there because we've removed the subpages from the nav. Is it possible to make those collapsible?
 - Fixed, these are now collapsible

It's hard to find the pages under About. Those pages are important, like Blogs, Talks, Community, Releases, etc.
   - These are now included in the left nav
   
What does search do? When I search for something like write.format.default, there isn't much context, only a list of pages that look unrelated. The pages also show up in the nav and push everything down.
 - Wow...thanks for pointing this out! I looked into this and it turns out this theme is using [flexsearch](https://github.com/nextapps-de/flexsearch) which is a pretty advanced full-text search library with some scoring algorithm that's doing way too much to predict relevance (and clearly missing the mark). Hugo already provides a way to generate a search index using the templates functionality (see `search-data.json`). I replaced all of the flexsearch code with a simple lookup of the search box's value to the content and a few lines of javascript to show a preview of where the match is and to link to the page. It looks to be working well and it lets us remove the flexsearch dependency.

Clicking the Iceberg logo in the upper left in /docs brings you back to the docs splash page, not the Iceberg splash site.
 - Fixed!

The splash site uses "Mailing lists" instead of "Community". Could you link to the community page instead?
 - Fixed!

Github and Slack buttons should open in a new page instead of loading in the current page.
 - Fixed!

On the splash page, I think we should add links to the features that are highlighted. We should also consider what to put here. The 3 listed don't seem compelling enough.
   - Great idea, I'll add this as I work on a more compelling feature set to display

@jackye1995 
for the issue that there is no link for those top level tabs like "Integration", I think we can just disable the link? Because the 2 that has link (Spark and Flink) are all linking to the getting started page, I think we can make them just a new sub-tab called Getting started and disable all top level tab links.
 - Done! Collapsing the lower sections freed up a lot of space to add this in it's on line.

Once I am in the doc page, I cannot go back to the landing page anymore, can we at least have a way to go back?
 - Fixed this by changing the Iceberg logo to return to the splash page like @rdblue suggested

the time travel script is setting a table property to perform time-travel, can we use the latest time travel syntax AS OF?
 - Will do! What's the Spark and Iceberg versions that support this? Wondering if we need to note that.

also agree that feature list is not full, we can think about the list of features together
  - Sounds good!

rdblue commented:
Thanks, @samredai! The site looks much better.

Before, we talked about having some pages that are not versioned and have just one copy throughout. Things like the Releases page (since that states the current release and is stale in 0.12.0) and the Spec would benefit from this. Is this possible?

Otherwise, I think the only remaining thing to fix is the splash page. Maybe we should merge this and then work on those two issues? @jackye1995 what do you think? Are we ready to move forward with this since it's in a separate repo and make improvements in parallel PRs?

samredai commented:
Done! Here's what I came up with:
- Moved all of the common pages to `docs/commons`. This includes the Community, Releases, and Format sections but let me know if there's any other pages (we can get as granular as a single markdown file).
- Added a [copy_to_docs.sh](https://github.com/apache/iceberg-docs/blob/c657a5566c5496178efe573d4e94225cb4d80b0d/docs/common/copy_to_docs.sh) script that copies everything in `docs/common` into `docs/content`.
- Added a step in the [deployment workflow](https://github.com/apache/iceberg-docs/blob/c657a5566c5496178efe573d4e94225cb4d80b0d/.github/workflows/deploy.yml#L85) to run the `copy_to_docs.sh` script during the `deploy-docs` job.

@rdblue @jackye1995 let me know what you think of this setup and I can add a section to the README explaining this.

jackye1995 commented:
@rdblue yes I totally agree we can move forward and make incremental progress on this, that also makes the review much easier with more specific changes to look at.

kbendick commented on .github/workflows/deploy.yml:
```diff
@@ -0,0 +1,98 @@
+    runs-on: ubuntu-20.04
+    steps:
+      - uses: actions/checkout@v2
+
```
Question: Is this `uses` supposed to be part of the same block as `- name: Set output`?

rdblue commented:
@samredai, if I understand the workflow correctly, it sounds like you're making extra copies of the common pages, one in each set of docs?

I think that still has a problem because there isn't a trigger that will rebuild the old docs branches when there is an update to the main content right?

What about just making those pages part of the splash site at the top level and linking to them from all the docs branches?

samredai commented:
@rdblue Yes you're understanding is correct. That's a good point, updating the common pages would still require redeploying the previous sites (even though the same document is deployed). Let me move this over to the landing-page site. It'll require a little extra designing since it will be out of the docs site and the landing-page site doesn't have any single page templates.

I'll put together a simple page template that matches the font and color of the main site, shows the markdown content, and has a simple set of links to either the home page (landing-page) or the docs site. That would let us do what you described, keep it as part of the splash site build and link to it from the docs branches!

rdblue commented:
The new top level pages look great. There are a few things still to fix, but I think we should commit this and fix those in parallel. @jackye1995 what do you think?

jackye1995 reviewed (approved):
Yes agree, I checked the demo website and it looks good to me overall. I will post issues I found after this is merged.

rdblue commented:
Thanks, @samredai! I merged this. And thanks to @jackye1995 for reviewing!
